src/breadcrumb-group/implementation.tsx

```diff
@@ -106,10 +126,45 @@ export function BreadcrumbGroupImplementation<T extends BreadcrumbGroupProps.Ite
     checkSafeUrl('BreadcrumbGroup', item.href);
   }
   const baseProps = getBaseProps(props);
-  const isMobile = useMobile();
+  const [navWidth, navRef] = useContainerQuery<number>(rect => rect.contentBoxWidth);
+  const mergedRef = useMergeRefs(navRef, __internalRootRef);
+
+  const itemsRefs = useRef<ItemsRefsType>({ ghost: {}, real: {} });
+  const setBreadcrumb = (type: keyof ItemsRefsType, index: string, node: null | HTMLLIElement) => {
+    if (node) {
+      itemsRefs.current[type][index] = node;
+    } else {
+      delete itemsRefs.current[type][index];
+    }
+  };
+
+  const [itemsWidths, setItemsWidths] = useState<ItemsWidthsType>({ ghost: [], real: [] });
+
+  useEffect(() => {
+    if (itemsRefs.current) {
+      const newItemsWidths: ItemsWidthsType = { ghost: [], real: [] };
+      for (const node of Object.values(itemsRefs.current.ghost)) {
+        const width = getLogicalBoundingClientRect(node).inlineSize;
+        newItemsWidths.ghost.push(width);
+      }
+      for (const node of Object.values(itemsRefs.current.real)) {
+        const width = getLogicalBoundingClientRect(node).inlineSize;
+        newItemsWidths.real.push(width);
+      }
+      if (
+        !areArrayEqual(newItemsWidths.ghost, itemsWidths.ghost) ||
+        !areArrayEqual(newItemsWidths.real, itemsWidths.real)
+      ) {
+        setItemsWidths(newItemsWidths);
+      }
+    }
+  }, [itemsWidths, items, navWidth]);
```

Comment:
You have a double render loop here. After calling `setItemsWidth`, this effect is called second time, because it tracks `itemsWidths` value which got just changed.

Reply:
Suggested solution


```js
setItemsWidths(oldWidths => {
  if (
    !areArrayEqual(newItemsWidths.ghost, oldWidths.ghost) ||
    !areArrayEqual(newItemsWidths.real, oldWidths.real)
  ) {
    return newItemsWidths;
  } else {
    return oldWidths
  }
})
```